artifacts/AdditionalSetupArtifacts.ps1

```diff
@@ -409,9 +412,7 @@ if (($env:cosmoServiceRestart -eq $false) -and ![string]::IsNullOrEmpty($env:saa
         -Force
 
     Write-Host " - Syncing all apps"
-    for ($i = 0; $i -lt 10; $i++) {
-        Get-NAVAppInfo -ServerInstance $ServerInstance -Tenant $tenantId | Sync-NAVApp -ServerInstance $ServerInstance -Tenant $tenantId -ErrorAction silentlycontinue -WarningAction silentlycontinue
-    }
+    Get-NAVAppInfo -ServerInstance $ServerInstance -Tenant $tenantId | Sync-NAVApp -ServerInstance $ServerInstance -Tenant $tenantId -ErrorAction silentlycontinue -WarningAction silentlycontinue
```

Comment:
Why did you remove the retry?

Reply:
While investigating a recent ticket, I wondered why the sync of the apps took so long, and I didn't saw the need to loop over all apps 10 times?

Reply:
We had some issues in the past where the sync didn't work on first try, but just retrying sometimes solved it. A better approach would probably be to check whether there are any out of sync apps and only then retry?

Reply:
I don't think there is a command or parameter to check that. What I can think of is tracking the apps that return a warning "...already synchronized..." and exclude them from the loop, but I'm not really a fan of that solution. I tested again, and the sync command executes nearly instantly for the already synced apps, so I think we can keep it as it was.

Reply:
I agree, performance-wise it shouldn't make a big difference, but if you want to make the effort, then `Get-NAVAppInfo` with `-TenantSpecificProperties` does the trick

```
PS C:\run> Get-NAVAppInfo -ServerInstance BC -Tenant Default -TenantSpecificProperties | ft

ServerInstance                ExtensionDataVersion IsInstalled SyncState NeedsUpgrade IsPublished AppId                                Name                            Publisher Version      
--------------                -------------------- ----------- --------- ------------ ----------- -----                                ----                            --------- -------
MicrosoftDynamicsNavServer$BC 23.6.18013.0                True    Synced        False        True 9856ae4f-d1a7-46ef-89bb-6ef056398228 System Application Test Library Microsoft 23.6.18013.0
MicrosoftDynamicsNavServer$BC 23.61.24400.0               True    Synced        False        True 3e745006-8ef7-4c52-94a5-e80ba883061b 4PS Scheduler (GRP) W1          4PS B.V.  23.61.24400.0
MicrosoftDynamicsNavServer$BC 23.61.24197.1               True    Synced        False        True a091c3a3-c4eb-4ecc-83f5-e95669ceb8bf 4PS System Application W1       4PS B.V.  23.61.24197.1
MicrosoftDynamicsNavServer$BC 23.6.18013.0                True    Synced        False        True 5d86850b-0d76-4eca-bd7b-951ad998e997 Tests-TestLibraries             Microsoft 23.6.18013.0
MicrosoftDynamicsNavServer$BC 23.6.0.0                    True    Synced        False        True c3735798-c911-415c-82ac-958a6d7b15da Container initializer           4PS B.V.  23.6.0.0
MicrosoftDynamicsNavServer$BC 23.6.0.0                    True    Synced        False        True 88573c9d-8c3c-467d-807e-8476c43a03bf Application                     4PS B.V.  23.6.0.0
MicrosoftDynamicsNavServer$BC 23.6.18013.0                True    Synced        False        True 40860557-a18d-42ad-aecb-22b7dd80dc80 Permissions Mock                Microsoft 23.6.18013.0
MicrosoftDynamicsNavServer$BC 23.61.24198.2               True    Synced        False        True 247e12d4-2649-4390-8349-240a67f10aa5 4PS Base Application W1         4PS B.V.  23.61.24198.2
MicrosoftDynamicsNavServer$BC 23.6.18013.0                True    Synced        False        True 5095f467-0a01-4b99-99d1-9ff1237d286f Library Variable Storage        Microsoft 23.6.18013.0
```

Actually that call with something like  `| Where-Object { $_.SyncState -ne "Synced" }` would probably be an easy yet efficient way

Reply:
Oh, thanks for pointing that out, I wasn't aware of the information this parameter provides. 👍
I have adjusted and tested it as you suggested.
Do you think this should be applied here as well:
https://github.com/cosmoconsult/container-scripts-for-selfservice/blob/fff4a7879fb4a87cb66b8ae26d7e2a5dd81dd3de/artifacts/AdditionalSetupArtifacts.ps1#L109-L116

Reply:
Yes, would absolutely make sense 👍

Reply:
done